src/gallia/transports/doip.py

```diff
@@ -661,17 +661,17 @@ async def write_request_raw(self, hdr: GenericHeader, payload: DoIPOutData) -> N
                 match payload:
                     case DiagnosticMessage():
                         # Now an ACK message is expected.
-                        await asyncio.wait_for(
-                            self._read_ack(payload.UserData),
+                        async with asyncio.timeout(
                             TimingAndCommunicationParameters.DiagnosticMessageMessageAckTimeout
                             / 1000,
-                        )
+                        ):
+                            await self._read_ack(payload.UserData)
                     case RoutingActivationRequest():
-                        await asyncio.wait_for(
-                            self._read_routing_activation_response(),
-                            TimingAndCommunicationParameters.RoutingActivationResponseTimeout
+                        async with asyncio.timeout(
+                            TimingAndCommunicationParameters.DiagnosticMessageMessageAckTimeout
```

Comment:
We should avoid introducing new bugs, especially with such seemingly straightforward changes...